driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -103,30 +109,35 @@ public InternalConnection get() {
 
     @Override
     public InternalConnection get(final long timeout, final TimeUnit timeUnit) {
+        PooledConnection pooledConnection;
         try {
             if (waitQueueSize.incrementAndGet() > settings.getMaxWaitQueueSize()) {
                 throw createWaitQueueFullException();
             }
             try {
-                connectionPoolListener.waitQueueEntered(new ConnectionPoolWaitQueueEnteredEvent(serverId));
-                PooledConnection pooledConnection = getPooledConnection(timeout, timeUnit);
-                if (!pooledConnection.opened()) {
-                    try {
-                        pooledConnection.open();
-                    } catch (Throwable t) {
-                        pool.release(pooledConnection.wrapped, true);
-                        if (t instanceof MongoException) {
-                            throw (MongoException) t;
-                        } else {
-                            throw new MongoInternalException(t.toString(), t);
-                        }
+                pooledConnection = getPooledConnection(timeout, timeUnit, true);
+            } catch (Throwable t) {
+                emitCheckOutFailedEvent(t);
+                throw (MongoException) t;
+            }
+            if (!pooledConnection.opened()) {
+                try {
+                    pooledConnection.open();
+                } catch (Throwable t) {
+                    pool.release(pooledConnection.wrapped, true);
+                    connectionPoolListener.connectionCheckOutFailed(new ConnectionCheckOutFailedEvent(serverId,
+                            Reason.CONNECTION_ERROR));
+                    if (t instanceof MongoException) {
```

Comment:
Replace this if-else block with just

    throw t;

I didn't realize that would work, but it does, and is simpler